File: app/src/main/java/com/stocky/Adapters/HistoryAdapter.java
package com.stocky.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.stocky.Model.History;
import com.stocky.R;

import java.util.ArrayList;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {

    private final ArrayList<History> histories;

    public HistoryAdapter(ArrayList<History> histories) {
        this.histories = histories;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.item_historico, parent, false);
        ViewHolder vh = new ViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        final String product = histories.get(position).getProduct();
        final String qtdEstoque = String.valueOf(histories.get(position).getQtd());
        final String dateStockChange = histories.get(position).getDate();
        final String tipoTransacao = histories.get(position).getTipoTransacao();

        holder.nomeProduto.setText(product);
        holder.qtdProduto.setText(qtdEstoque);
        holder.dateChange.setText(dateStockChange);

        if (tipoTransacao.equals("entrada")) {
            holder.nomeProduto.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.green));
            holder.qtdProduto.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.green));
            holder.dateChange.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.green));
        } else if (tipoTransacao.equals("saida")){
            holder.nomeProduto.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.red));
            holder.qtdProduto.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.red));
            holder.dateChange.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.red));
        }
    }

    @Override
    public int getItemCount() {
        return histories.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView nomeProduto;
        public TextView qtdProduto;
        public TextView dateChange;
        public View layout;

        public ViewHolder(View itemView) {
            super(itemView);
            layout = itemView;
            nomeProduto = itemView.findViewById(R.id.nomeProduto);
            qtdProduto = itemView.findViewById(R.id.qtdEstoque);
            dateChange = itemView.findViewById(R.id.dateChamge);
        }
    }
}

File: app/src/main/java/com/stocky/adapters/HistoryAdapter.kt
package com.stocky.adapters

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.stocky.R
import android.widget.TextView
import com.stocky.model.History
import java.util.ArrayList

class HistoryAdapter(private val histories: ArrayList<History?>) :
    RecyclerView.Adapter<HistoryAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val v = inflater.inflate(R.layout.item_history, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val product = histories[position]?.product.toString()
        val stockQuantity = histories[position]?.qtd.toString()
        val dateStockChange = histories[position]?.date
        val transactionType = histories[position]?.tipoTransacao
        holder.productName.text = product
        holder.productQuantity.text = holder.productQuantity.context.getString(R.string.product_quantity, stockQuantity)
        holder.dateChange.text = holder.dateChange.context.getString(R.string.product_date_change, dateStockChange)
        if (transactionType == KEY_TRANSACTION_STOCK_IN) {
            holder.iconUpDown.setImageResource(R.drawable.ic_arrow_up)

        } else if (transactionType == KEY_TRANSACTION_STOCK_OUT) {
            holder.iconUpDown.setImageResource(R.drawable.ic_arrow_down)
        }
    }

    override fun getItemCount(): Int {
        return histories.size
    }

    class ViewHolder(var layout: View) : RecyclerView.ViewHolder(
        layout
    ) {
        var productName: TextView = itemView.findViewById(R.id.productName)
        var productQuantity: TextView = itemView.findViewById(R.id.stockQuantity)
        var dateChange: TextView = itemView.findViewById(R.id.dateChange)
        var iconUpDown : com.google.android.material.imageview.ShapeableImageView = itemView.findViewById(R.id.iconUpDown)

    }

    companion object {
        private const val KEY_TRANSACTION_STOCK_IN = "entrada"
        private const val KEY_TRANSACTION_STOCK_OUT = "saida"
    }
}